synapse/storage/_base.py

```diff
@@ -114,7 +114,7 @@ def db_to_json(db_content: Union[memoryview, bytes, bytearray, str]) -> Any:
         db_content = db_content.tobytes()
 
     # Decode it to a Unicode string before feeding it to the JSON decoder, since
-    # Python 3.5 does not support deserializing bytes.
+    # it only supports handling strings
```

Comment:
Isn't the opposite true now? That Python > 3.5 supports bytes or strings?

Reply:
There's a bit of an explanation in the commit message of 0506cc9. Essentially we switched to an instance of `JSONDecoder` that looks to only support `str` arguments now. The comment was originally written when we were using `json.decode`, which did support different argument types based on the Python version.